src/backend/distributed/executor/executor.c

```diff
@@ -846,6 +856,13 @@ AssignTasksToConnections(DistributedExecution *execution)
 				connection =
 					StartPlacementListConnection(connectionFlags, placementAccessList,
 												 NULL);
+
+				/*
+				 * We'll assign tasks almost at the same time, in the loop right now.
+				 * So, it is acceptable to have a tiny difference between the first and
+				 * the last assignments.
+				 */
+				workerPool->connectionStart = connection->connectionStart;
```

Comment:
I think this is the right place to put this, but wondering whether we should only do this when we first establish a new connection. Right now it probably doesn't matter, but if we keep adding a new connection every 10ms the time-out will keep resetting.